### PR TITLE
karma update messages to dedicated log channel + remove SPECIAL_REPLIES

### PR DESCRIPTION
**Pull request by bbelderbos**

### CODE REVIEW

File: src/karmabot/db/karma_note.py [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
+
+from karmabot.db.modelbase import SqlAlchemyBase
+
+
+class KarmaNote(SqlAlchemyBase):
+    """Models a simple note system in the DB"""
+
+    __tablename__ = "karma_note"
+
+    id = sa.Column(  # noqa
```
Comment (bbelderbos) [inline src/karmabot/db/karma_note.py:13]: mypy was not happy about this

File: src/karmabot/settings.py [hunk 2]
```diff
@@ -2,11 +2,12 @@
 import os
 import re
 from pathlib import Path
+from typing import Union
 
 from dotenv import load_dotenv
 
 
-def get_env_var(env_var: str, default: str | None = None) -> str:
+def get_env_var(env_var: str, default: Union[str, None] = None) -> str:
```
Comment (bbelderbos) [inline src/karmabot/settings.py:10]: weird although I was using Python 3.10 mypy (in pre-commit) said | was not ok so had to change to Union to even be able to commit

File: src/karmabot/bot.py [hunk 3]
```diff
@@ -157,18 +143,6 @@ def reply_help(message, say):
     say(text=text, channel=user_id)
 
 
-# Message replies
-@app.message(SPECIAL_WORDS_PATTERN)  # type: ignore
```
Comment (bbelderbos) [inline src/karmabot/bot.py:161]: https://github.com/PyBites-Open-Source/karmabot/issues/113

**Comment by bbelderbos** [comment]
Not sure why more commits show up but the diff is accurate. I tested it in UAT and works nicely:

Screenshot 2023-07-28 at 19 05 09

Screenshot 2023-07-28 at 19 05 19

Screenshot 2023-07-28 at 19 05 42

Screenshot 2023-07-28 at 19 05 45

Screenshot 2023-07-28 at 19 05 50

**Comment by bbelderbos** [comment]
@JnyJny logging the channel would be nice, but given current way the code is structured that would require a bigger change I think.

**Comment by bbelderbos** [comment]
> @JnyJny logging the channel would be nice, but given current way the code is structured that would require a bigger change I think.

Although we still can if we pass down channel_id into `karma.change_karma(...` and there pass it through to `self._create_msg(..` to make it part of `text` there, but not sure how useful a Cxyz channel code is (I don't think you get the name of the channel)

File: src/karmabot/settings.py (hunk 2 above)
Comment (JnyJny) [reply src/karmabot/settings.py:10]: Might have wanted `Optional`? Not sure and it's a mighty fine hair to split.  LGTM.

**Comment by JnyJny** [comment]
Presuming everything tests ok (did not check) LGTM!

**Review by JnyJny — APPROVED** [review]
LGTM

**Comment by pmayd** [comment]
OK you beat me to it!

File: src/karmabot/settings.py (hunk 2 above)
Comment (bbelderbos) [reply src/karmabot/settings.py:10]: Following Sebastian's opinion here ;)
https://fastapi.tiangolo.com/python-types/#using-union-or-optional

**Pull request merged**